ush/python/pygfs/task/atmens_analysis.py

```diff
@@ -0,0 +1,332 @@
+
+
+class AtmEnsAnalysis(Analysis):
+    """
+    Class for global atmens analysis tasks
+    """
+    @logit(logger, name="AtmEnsAnalysis")
+    def __init__(self, config):
+        super().__init__(config)
+
+        _res = int(self.config.CASE_ENKF[1:])
+        _res_anl = int(self.config.CASE_ANL[1:])
+        _window_begin = add_to_datetime(self.runtime_config.current_cycle, -to_timedelta(f"{self.config.assim_freq}H") / 2)
+        _fv3jedi_yaml = os.path.join(self.runtime_config.DATA, f"{self.runtime_config.CDUMP}.t{self.runtime_config.cyc:02d}z.atmens.yaml")
+
+        # Create a local dictionary that is repeatedly used across this class
+        local_dict = AttrDict(
+            {
+                'npx_ges': _res + 1,
+                'npy_ges': _res + 1,
+                'npz_ges': self.config.LEVS - 1,
+                'npz': self.config.LEVS - 1,
+                'npx_anl': _res_anl + 1,
+                'npy_anl': _res_anl + 1,
+                'npz_anl': self.config.LEVS - 1,
+                'ATM_WINDOW_BEGIN': _window_begin,
+                'ATM_WINDOW_LENGTH': f"PT{self.config.assim_freq}H",
+                'comin_ges_atm': self.config.COMIN_GES,
+                'comin_ges_atmens': self.config.COMIN_GES_ENS,
+                'OPREFIX': f"{self.config.EUPD_CYC}.t{self.runtime_config.cyc:02d}z.",  # TODO: CDUMP is being replaced by RUN
+                'APREFIX': f"{self.runtime_config.CDUMP}.t{self.runtime_config.cyc:02d}z.",  # TODO: CDUMP is being replaced by RUN
+                'GPREFIX': f"gdas.t{self.runtime_config.previous_cycle.hour:02d}z.",
+                'fv3jedi_yaml': _fv3jedi_yaml,
+            }
+        )
+
+        # task_config is everything that this task should need
+        self.task_config = AttrDict(**self.config, **self.runtime_config, **local_dict)
+
+    @logit(logger)
+    def initialize(self: Analysis) -> None:
+        """Initialize a global atmens analysis
+
+        This method will initialize a global atmens analysis using JEDI.
+        This includes:
+        - staging CRTM fix files
+        - staging FV3-JEDI fix files
+        - staging model backgrounds
+        - generating a YAML file for the JEDI executable
+        - linking the JEDI executable (TODO make it copyable, requires JEDI fix)
+        - creating output directories
+
+        Parameters
+        ----------
+        Analysis: parent class for GDAS task
+
+        Returns
+        ----------
+        None
+        """
+        super().initialize()
+
+        # stage CRTM fix files
+        crtm_fix_list_path = os.path.join(self.task_config.HOMEgfs, 'parm', 'parm_gdas', 'atm_crtm_coeff.yaml')
+        logger.debug(f"Staging CRTM fix files from {crtm_fix_list_path}")
+        crtm_fix_list = parse_yamltmpl(crtm_fix_list_path, self.task_config)
+        FileHandler(crtm_fix_list).sync()
+
+        # stage fix files
+        jedi_fix_list_path = os.path.join(self.task_config.HOMEgfs, 'parm', 'parm_gdas', 'atm_jedi_fix.yaml')
```

Comment:
and this too.  
we should ask if we will need different yaml's for crtm, jedi_fix, etc. between `var` and `ens` modes?

Reply:
See comment above for staging of crtm files.  If the changes above are acceptable, I'll remove the jedi and crtm fix file staging from `atm_analysis.py`.

Reply:
Test implementation in working copy of `RussTreadon-NOAA:feature/refactor_jediens` does not cover UFS-DA Aerosol case.   Need to take a step back and think more holistically.   Generalization will likely require changes to UFS-DA Aerosl, ATM, and ENS applications.   Would be good to capture this in a new issue and subsequent PR.

Reply:
Recommend that unification of fix file staging across UFS-DA applications be the subject of a new g-w issue.   What do you think @aerorahul and @CoryMartin-NOAA ?